### ISSUE TITLE
PrintWifi and 802.11ac

### ISSUE THREAD

**Issue by bjooris** [kohler/click#172]
Hi all,

I'am trying to use click on top of an ath10k compliant card in monitor mode.
Wireshark is able to decode (radiotap header) my incoming frames transmitted from a station to an AP (two other devices).
My click script looks like this 
FromDevice(wlan1, PROMISC 1) -> RadiotapDecap() -> PrintWifi() -> Discard();

Some parts look ok like tods, data,... but some values like rate, rssi,.. are not being decoded.
 
2008 |  0Mb + 0/ 0 | data tods 
2008 |  0Mb + 0/ 0 | data tods 
....

radiotap.h and RadiotapDecap.cc seems to be changed recently.

Can you suggest how I can try to fix this?

Thanks a lot in advance,

Bart

**Comment by rommon** [comment]
Hi,

I guess it is the same problem as I described in #180 . Could you if it is possible append a wireshark trace as the interface is in monitor mode?


BR,
rommon

**Comment by bjooris** [comment]
Hi Rommon,

I don't think the pcap file in attachment has the " multiple it_present 
fields" as you mentioned in #180 
<https://github.com/kohler/click/issues/180>.
But I would like to try your patch. If it doesn't fix our issue I'm sure 
it will be good starting point.

Thanks!

Bart




On 03/14/2015 05:43 PM, Rom wrote:
>
> Hi,
>
> I guess it is the same problem as I described in #180 
> <https://github.com/kohler/click/issues/180> . Could you if it is 
> possible append a wireshark trace as the interface is in monitor mode?
>
> BR,
> rommon
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/kohler/click/issues/172#issuecomment-80568952>.
>


-- 
Bart Jooris
Department of Information Technology
Internet Based Communication Networks and Services (IBCN)
Ghent University - iMinds
Gaston Crommenlaan 8 (Bus 201), B-9050 Gent, Belgium
E-mail: bart.jooris@intec.UGent.be
M: +32 474 59 53 42
Tel. +32 9 33 14900
Fax +32 9 33 14899

**Comment by rommon** [comment]
please let me know if patch #183 is working for you

BR,
Rommon

**Comment by bjooris** [comment]
Hi Rommon,

after applying the patch, RadiotapDecap fails to compile.
Would it be correct to append your calculated additional_it_present_flags to the params of rt_check_header on line 133 (after your patch)?

Thanks,

Bart

**Comment by rommon** [comment]
yes, you are right. I must have forgotten this parameter. I need to fix that in my pullrequest, not sure how to do that actually.

**Comment by rommon** [comment]
ok, added it to the pull request.
